Epilepsy/scripts/epilepsy-labeler.py

```diff
@@ -0,0 +1,323 @@
+sv = SentenceNgramViewer(candidates, session)
+
+
+# In[ ]:
+
+sv
+
+
+# # Label Functions
+
+# Here is the fundamental part of the project. Below are the label functions that are used to give a candidate a label of 1,0 or -1 which corresponds to correct relation, not sure and incorrection relation. The goal here is to develop functions that can label as many candidates as possible.
+
+# In[ ]:
+
+import re
+from snorkel.lf_helpers import (
+    get_left_tokens,
+    get_right_tokens, 
+    get_between_tokens,
+    get_tagged_text,
+    get_text_between,
+    rule_regex_search_tagged_text,
+    rule_regex_search_btw_AB,
+    rule_regex_search_btw_BA,
+    rule_regex_search_before_A,
+    rule_regex_search_before_B,
+)
+
+
+# In[ ]:
+
+import pandas as pd
+gene_list = pd.read_csv('epilepsy-genes.tsv',sep="\t")
+
+
+# In[ ]:
+
+variation_words = {"mutation", "mutation-negative", "mutation-positive",
+                   "de novo", "heterozygous", "homozygous", "deletion", "variants","variant",
+                   "recessive","autosomal", "haploinsufficiency", "knock-out", "genotype",
+                  "null"}
+
+cause_words = {"cause", "caused", "maps", "due", "associated"}
+neg_words = {"serum", "intervention", "negative"}
+
+related_diseases_symptoms = {"epileptic encephalopathies","epileptic encephalopathy", "seizures", "encephalopathy",
+                             "epileptic spasms", "myoclonic astatic epilepsy", "neurodevelopmental",
+                            "refractory epilepsy", "severe myoclonic epilepsy of infancy",
+                            "dravet syndrome", "myoclonic-astatic epilepsy", "absence epilepsy", 
+                            "epilepsies", "west syndrome", "seizures", "autoimmune epilepsy",
+                            "temporal libe epilepsy"}
+
+unrelated_diseases = {"continguous gene syndrome", "X-linked clinical syndrome", 
+                      "insulin-dependent diabetes mellitus", "stiff-person syndrome",
+                     "vascular syndrome","autophagic vacuolar myopathy","cardiomyopathy",
+                     "Chinese linear nevus sebaceous syndrome", "MPPH syndrome",
+                     "hypertrophic cardiomyopathy", "mowat-wilson syndrome", "hunter syndrome",
+                     "Nephrotic Syndrome", "Vici syndrome"}
+
+disese_abbreviations_pos = {"SMEI", "DS", "MAE", "CS", "TLE", "ADLTE", "EIEE"}
+disease_abbreviations_neg = {"MWS", "KDT", "TSC", "CCHS", "IDDM", "BPP", "NS"}
+
+gene_adj = {"-related", "anti-", "-gene"}
+model_organisms = {"mice", "zebrafish", "drosophila"}
+
+disease_context = {"patients with", "individuals with", "cases of", "treatment of"}
+
+# IF {{B}}} {{A}} or vice versa then not a valid relationship
+def LF_abbreviation(c):
+    if len(get_text_between(c)) < 3:
+        return -1
+    return 0
+
+# IF {{a}} is a {{B}} or {{B}} is a {{A}}
+def LF_is_a(c):
+    return rule_regex_search_btw_AB(c,r'.* is a .*',-1) or rule_regex_search_btw_BA(c,r'is a',-1)
```

Comment:
what is this supposed to do? the or statement will return the first condition if that evaluates to True (i.e. not `0`). Otherwise will return second condition. Is this intended?